File: ABC253/probC.py
ms = {}
for i in range(int(input())):
    query = list(input().split())
    if query[0] == '1':
        ms[query[1]] = ms.get(query[1], 0) + 1
    elif query[0] == '2':
        ms[query[1]] -= min(int(query[2]), ms.get(query[1], 0))
    elif query[0] == '3':
        msList = [int(key) for key, val in ms.items() if val > 0]
        print(max(msList) - min(msList))

File: ABC253/probC.cpp
#include <bits/stdc++.h>

int main(){
    long long Q;
    std::cin >> Q;
    std::string query;
    std::multiset<long long> set;
    for (long long i = 0; i < Q; i++)
    {
        std::cin >> query;
        if (query[0] == '1')
        {
            set.insert((long long)query[2]);
        }
        else if (query[0] == '2')
        {
            long long search = (long long)query[2], fix = (long long)query[4];
            long long count = std::min(fix, (long long)set.count(search));
            for (long long j = 0; j < count; j++)
            {
                set.erase(set.find(search));
            }
        }
        else
        {
            long long max = 0;
            long long min = 9223372036854775800;
            for (long long val : set){
                max = std::max(max, val);
                min = std::min(min, val);
            }
            std::cout << max - min << std::endl;
        }
    }
}